### PR TITLE
replacing intercom with helpscout

### LINKED ISSUE #7863: [CAL-1309] replace helpscout with intercom
[https://www.intercom.com/help/en/articles/168-install-intercom-in-your-product-for-logged-in-users](https://www.intercom.com/help/en/articles/168-install-intercom-in-your-product-for-logged-in-users)

we are using this package: [https://github.com/calibreapp/react-live-chat-loader#-supported-providers](https://github.com/calibreapp/react-live-chat-loader#-supported-providers)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1309](https://linear.app/calcom/issue/CAL-1309/replace-helpscout-with-intercom)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
TODO:

- [x] add staging and preview environments
- [x] add intercom key in production environments
- [x] remove helpscout key in production environments
- [x] [sync with userdata
](https://github.com/calcom/cal.com/pull/7865#discussion_r1143668465)
- [x] [open chat when clicking "Contact Support"
](https://github.com/calcom/cal.com/pull/7865#discussion_r1143670395)

CleanShot 2023-03-21 at 17 18 58@2x

### CODE REVIEW

File: apps/web/pages/_app.tsx [hunk 1]
```diff
@@ -25,6 +32,15 @@ const calFont = localFont({
 });
 
 function MyApp(props: AppProps) {
+  useEffect(() => {
+    (window as any).intercomSettings = {
+      api_base: "https://api-iam.intercom.io",
+      name: "Jane Doe", // Full name – doesn't work yet
+      email: "customer@example.com", // Email address – doesn't work yet
+      created_at: 1312182000, // Signup date as a Unix timestamp – doesn't work yet
```
Comment (PeerRich) [inline apps/web/pages/_app.tsx:40]: this needs to be fixed and replaced with actual user data. couldn't find a way to make this work yet (not even with Jane Doe sample data)

File: package.json [hunk 2]
```diff
@@ -93,6 +93,7 @@
   "dependencies": {
     "city-timezones": "^1.2.1",
     "eslint": "^8.34.0",
+    "react-live-chat-loader": "^2.8.1",
```
Comment (PeerRich) [inline package.json:96]: not sure if we should import it here or in a different package

File: packages/features/ee/support/components/HelpMenuItem.tsx [hunk 3]
```diff
@@ -210,6 +210,7 @@ export default function HelpMenuItem({ onHelpItemSelect }: HelpMenuItemProps) {
             } else {
               loadChat({ open: true });
             }
+            // TODO: somehow intercom doensnt open yet
```
Comment (PeerRich) [inline packages/features/ee/support/components/HelpMenuItem.tsx:213]: not sure why but somehow intercom doesn't open when clicking. it should tho

**Comment by G3root** [comment]
@PeerRich ready for review

[Screencast 2023-03-22 01:31:51.webm](https://user-images.githubusercontent.com/84864519/226728743-58ede9d5-1347-4de7-97e5-02d66800c8e2.webm)

File: packages/features/ee/support/lib/intercom/useIntercom.ts [hunk 4]
```diff
@@ -1,12 +1,35 @@
...
+export const useIntercom = () => {
+  const hookData = useIntercomHook();
+  const { data } = trpc.viewer.me.useQuery();
+
+  const open = () => {
+    hookData.boot({
+      name: data?.name ?? "",
+      email: data?.email,
+      userId: String(data?.id),
+      createdAt: String(dayjs().unix()),
```
Comment (PeerRich) [inline packages/features/ee/support/lib/intercom/useIntercom.ts:28]: this should be the day the user was created, not today

File: packages/features/ee/support/lib/intercom/useIntercom.ts [hunk 5]
```diff
@@ -1,12 +1,50 @@
...
+      email: data?.email,
+      userId: String(data?.id),
+      createdAt: String(dayjs(data?.createdDate).unix()),
+      customAttributes: {
+        //keys should be snake cased
+        user_name: data?.username,
+        identity_provider: data?.identityProvider,
+        timezone: data?.timeZone,
+        locale: data?.locale,
+        completed_onboarding: data?.completedOnboarding,
```
Comment (emrysal) [inline packages/features/ee/support/lib/intercom/useIntercom.ts:39]: I don't think it's possible to open the Help menu before onboarding is completed 🤔

Comment (PeerRich) [inline packages/features/ee/support/lib/intercom/useIntercom.ts:39]:
```suggestion
```

File: packages/features/ee/support/lib/intercom/useIntercom.ts [hunk 6]
```diff
@@ -1,12 +1,51 @@
...
+  const open = () => {
+    hookData.boot({
+      name: data?.name ?? "",
+      email: data?.email,
+      userId: String(data?.id),
+      createdAt: String(dayjs(data?.createdDate).unix()),
+      customAttributes: {
+        //keys should be snake cased
+        user_name: data?.username,
+        link: CAL_URL + "/" + data?.username,
```
Comment (PeerRich) [inline packages/features/ee/support/lib/intercom/useIntercom.ts:37]: @G3root fyi I added "link"

**Review by emrysal — APPROVED** [review]
Follow-up needed to implement this more performant with LiveChatLoaderProvider, as well as some fixes for self-hosting. Will tackle after and get this released.

**Pull request merged**